plugin/kubernetai/podhandler.go

```diff
@@ -1,21 +1,43 @@
 package kubernetai
 
 import (
+	"reflect"
+
 	"github.com/coredns/coredns/plugin/kubernetes"
 	"github.com/coredns/coredns/plugin/kubernetes/object"
 )
 
 type podHandlerItf interface {
-	PodWithIP(k kubernetes.Kubernetes, ip string) *object.Pod
+	PodWithIP(k *kubernetes.Kubernetes, ip string) *object.Pod
+	IsPodVerified(k *kubernetes.Kubernetes) bool
+}
+
+type podHandler struct {
+	IsVerifiedCache map[string]podVerified
 }
 
-type podHandler struct{}
+type podVerified struct {
+	isVerified  bool
+	initialized bool
+}
 
 // podWithIP return the api.Pod for source IP ip. It returns nil if nothing can be found.
-func (p *podHandler) PodWithIP(k kubernetes.Kubernetes, ip string) *object.Pod {
+func (p *podHandler) PodWithIP(k *kubernetes.Kubernetes, ip string) *object.Pod {
 	ps := k.APIConn.PodIndex(ip)
 	if len(ps) == 0 {
 		return nil
 	}
 	return ps[0]
 }
+
+func (p *podHandler) IsPodVerified(k *kubernetes.Kubernetes) bool {
+	cacheKey := k.Name()
+	cacheVal := p.IsVerifiedCache[cacheKey]
+	if cacheVal.initialized == false {
+		kubeReflection := reflect.ValueOf(k).Elem()
```

Comment:
You were saying that you wanted to remove `reflect` and instead expose `Kubernetes` `podMode` before submitting this upstream?

Reply:
Yeah, idealy we should be able to read the `PodMode` from the underlying kubernetes plugin.
It does require to update the kubernetes plugin to do so though.